internal/endtoend/endtoend_test.go

```diff
@@ -95,52 +110,103 @@ func TestReplay(t *testing.T) {
+						Managed: true,
+					}
+				}
+			},
+			Enabled: func() bool {
+				if len(os.Getenv("CI")) > 0 {
+					return false
+				}
+				return len(os.Getenv("SQLC_AUTH_TOKEN")) > 0
+			},
+		},
+	}
+
 	for _, replay := range dirs {
 		tc := replay
-		t.Run(tc, func(t *testing.T) {
-			t.Parallel()
+		for name, testctx := range contexts {
+			name := name
+			testctx := testctx
 
-			var stderr bytes.Buffer
-			var output map[string]string
-			var err error
+			if !testctx.Enabled() {
+				continue
+			}
 
-			path, _ := filepath.Abs(tc)
-			args := parseExec(t, path)
-			expected := expectedStderr(t, path)
+			t.Run(filepath.Join(name, tc), func(t *testing.T) {
+				t.Parallel()
+				var stderr bytes.Buffer
+				var output map[string]string
+				var err error
 
-			if args.Process != "" {
-				_, err := osexec.LookPath(args.Process)
-				if err != nil {
-					t.Skipf("executable not found: %s %s", args.Process, err)
+				path, _ := filepath.Abs(tc)
+				args := parseExec(t, path)
+
+				if args.Process != "" {
+					_, err := osexec.LookPath(args.Process)
+					if err != nil {
+						t.Skipf("executable not found: %s %s", args.Process, err)
+					}
 				}
-			}
 
-			env := cmd.Env{
-				Debug:    opts.DebugFromString(args.Env["SQLCDEBUG"]),
-				NoRemote: true,
-			}
-			switch args.Command {
-			case "diff":
-				err = cmd.Diff(ctx, env, path, "", &stderr)
-			case "generate":
-				output, err = cmd.Generate(ctx, env, path, "", &stderr)
-				if err == nil {
-					cmpDirectory(t, path, output)
+				if len(args.Contexts) > 0 {
+					if !slices.Contains(args.Contexts, name) {
+						t.Skipf("unsupported context: %s", name)
+					}
 				}
-			case "vet":
-				err = cmd.Vet(ctx, env, path, "", &stderr)
-			default:
-				t.Fatalf("unknown command")
-			}
 
-			if len(expected) == 0 && err != nil {
-				t.Fatalf("sqlc %s failed: %s", args.Command, stderr.String())
-			}
+				expected := expectedStderr(t, path)
+				opts := cmd.Options{
+					Env: cmd.Env{
+						Debug:    opts.DebugFromString(args.Env["SQLCDEBUG"]),
+						NoRemote: true,
+					},
+					Stderr:       &stderr,
+					MutateConfig: testctx.Mutate,
+				}
 
-			if diff := cmp.Diff(expected, stderr.String()); diff != "" {
-				t.Errorf("stderr differed (-want +got):\n%s", diff)
-			}
-		})
+				switch args.Command {
+				case "diff":
+					err = cmd.Diff(ctx, path, "", &opts)
+				case "generate":
+					output, err = cmd.Generate(ctx, path, "", &opts)
+					if err == nil {
+						cmpDirectory(t, path, output)
+					}
+				case "vet":
+					err = cmd.Vet(ctx, path, "", &opts)
+				default:
+					t.Fatalf("unknown command")
+				}
+
+				if len(expected) == 0 && err != nil {
+					t.Fatalf("sqlc %s failed: %s", args.Command, stderr.String())
+				}
+
+				var diff string
+				for _, expectedErr := range strings.Split(expected, "---\n") {
+					diff = cmp.Diff(strings.TrimSpace(expectedErr), strings.TrimSpace(stderr.String()))
+					if diff == "" {
+						break
+					}
+				}
```

Comment:
This isn't ideal, but might not be worth the effort to clean up. Just matching **some** stderr output is worse than matching the actual expected output right? There are a couple of ways I can think of to have the expected output depend on the context explicitly (separate files with magic names, or demarcating with `"--- <context-name>\n"`), but I'm wondering what you think.